Automattic/wp-calypso
# Reader: Remove unnecessary thumbnail request tracking

#### Changes proposed in this Pull Request

The `isRequestingThumbnailUrl` selector has been unused since its creation in #10001. When we remove it, we can also remove the `requesting` reader thumbnail sub-reducer, because it's unused as well. That sub-reducer is the only consumer of the `READER_THUMBNAIL_REQUEST`, `READER_THUMBNAIL_REQUEST_SUCCESS`, and `READER_THUMBNAIL_FAILURE` actions, so we can stop dispatching those as well. Finally, we can remove some related unit tests. 

This PR removes all of the aforementioned.

#### Testing instructions

* Verify tests pass: `yarn run test-client client/state/reader/thumbnails`
* Go to `/read`, search for `youtube`, scroll down and verify thumbnails still load correctly.
